### DISCUSSION TITLE
How can I set up view-specific controller behavior across multiple views?

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#7574] [Q&A] [answered]
Hi, I am experimenting with linking different views and using different view states here: https://observablehq.com/d/b4011e0e79c2d493. However, I would like to know if it is possible to set up unique controller behavior for different views - specifically I would like to have a controller set up with a zoomAxis of 'all' for one view, 'X' for another, and 'Y' for another.

Below is a snippet of code (see above link for the complete notebook on ObservableHQ) showing how I'm setting up the controller and experimenting with having different view states for different views. 

```
deckgl = {

  var ini_zoom_mode = 'Y'

  const instance = new deck.DeckGL({
    container,
    views,
    controller: {scrollZoom: true, inertia: true, zoomAxis: ini_zoom_mode},

    // both seem to work
    /////////////////////////
    // initialViewState: ini_view_state,
    viewState: ini_view_state,

    onViewStateChange: ({viewState, viewId}) => {

      if (viewId === 'left_heatmap') {
          
        viewState.left_heatmap = viewState;
        viewState.right_heatmap = ini_view_state_generic
      
        // console.log('left heatmap', viewState.right_heatmap)

      } else {
        
        // viewState.left_heatmap = { ...viewState.left_heatmap}

        viewState.left_heatmap = ini_view_state_generic
        viewState.right_heatmap = viewState;

        // try to set the left target
        viewState.left_heatmap.target = [ini_view_state_generic.target[0], viewState.target[1]]

        console.log('right heatmap', viewState.left_heatmap)
        
      }
      
      // Update deck
      instance.setProps({viewState: {...viewState}});
    }

```

The issue I'm seeing is that the contoller zoomAxis only seems to effect the left heatmap and not the right (see GIF).

![deckgl_controller](https://user-images.githubusercontent.com/8352840/211722195-2d5cc0e2-6368-4a23-85af-fb84ea845b21.gif)

Is there a way that I can set up multiple controllers or modify the controller behavior based on which view I am in?

**Comment by Pessimistress** [comment, marked as answer]
You should set the [controller option](https://deck.gl/docs/api-reference/core/view#controller) in each view's constructor. The top-level `controller` prop is just a shorthand when you use the default view, as explained [here](https://deck.gl/docs/api-reference/core/deck#controller).

**Reply by cornhundred** [reply]
Thanks that works - here is an updated link with X and Y zooming for each view: https://observablehq.com/d/8b00d401037776cd
